src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -906,6 +910,18 @@ public async Task HandleDidChangeTextDocumentNotification(ScriptFile[] changedFi
             }
         }
 
+        internal async Task HandleRefreshTokenNotification(
+            RefreshTokenParams refreshTokenParams,
+            EventContext eventContext
+        )
+        {
+            await Task.Run(() =>
```

Comment:
awaiting a Task.Run isn't really adding anything useful, you might as well get rid of it and just call `UpdateAuthToken` directly, then return `Task.CompletedTask;`